File: articles/linear-algebra-for-systems-modeling/articles/graph-theory-foundations-for-systems-modeling/calculators/r/graph_theory_calculator.R
result <- data.frame(
  calculator = "graph_theory_foundations_calculator",
  graph_name = "synthetic_infrastructure_graph_foundations",
  node_count = 5,
  edge_count = 6,
  directed = FALSE,
  weighted = TRUE,
  component_count = 1,
  max_degree = 3,
  min_degree = 2,
  average_degree = 2.4,
  has_cycle = TRUE,
  graph_density = 0.6,
  warning = "Graph metrics depend on node definitions, edge definitions, graph boundaries, weight semantics, temporal scope, and data provenance."
)

dir.create("outputs", recursive = TRUE, showWarnings = FALSE)
write.csv(result, "outputs/r_graph_theory_foundations_calculator.csv", row.names = FALSE)
print(result)

File: articles/linear-algebra-for-systems-modeling/articles/graph-theory-foundations-for-systems-modeling/calculators/python/graph_theory_calculator.py
from __future__ import annotations

import csv
import json
from pathlib import Path


def main() -> None:
    output_dir = Path("outputs")
    output_dir.mkdir(parents=True, exist_ok=True)

    result = {
        "calculator": "graph_theory_foundations_calculator",
        "graph_name": "synthetic_infrastructure_graph_foundations",
        "node_count": 5,
        "edge_count": 6,
        "directed": False,
        "weighted": True,
        "component_count": 1,
        "max_degree": 3,
        "min_degree": 2,
        "average_degree": 2.4,
        "has_cycle": True,
        "graph_density": 0.6,
        "warning": "Graph metrics depend on node definitions, edge definitions, graph boundaries, weight semantics, temporal scope, and data provenance."
    }

    with (output_dir / "graph_theory_foundations_calculator.json").open("w", encoding="utf-8") as handle:
        json.dump(result, handle, indent=2, sort_keys=True)

    with (output_dir / "graph_theory_foundations_calculator.csv").open("w", newline="", encoding="utf-8") as handle:
        writer = csv.DictWriter(handle, fieldnames=list(result.keys()))
        writer.writeheader()
        writer.writerow(result)

    print(json.dumps(result, indent=2, sort_keys=True))


if __name__ == "__main__":
    main()
